docs/tutorials/debugging-clojure-script.md

```diff
@@ -0,0 +1,161 @@
+# Debugging ClojureScript using dap-mode
+
+In this tutorial, we will cover the steps needed to debug `ClojureScript` in
+`emacs` using [Google Chrome Debug Adapter](https://github.com/microsoft/vscode-chrome-debug). For general overview of Clojure
+features check [Clojure Guide](clojure-guide.md).
+
+
+## Requirements
+* Emacs 26.1 or Emacs 27.1+ (recommended)
+* node.js (v6.0.0+, most recent version preferred)
+* npm (comes bundled with node.js) or yarn
+* Java SDK (Version 8+, Hotspot)
+* Google Chrome browser
+
+# Sample minimal configuration
+
+Here it is sample configuration based on the basic configuration in [Clojure Guide](clojure-guide.md).
+
+``` emacs-lisp
+(require 'package)
+(add-to-list 'package-archives '("melpa" . "http://melpa.org/packages/") t)
+(package-initialize)
+
+(setq package-selected-packages '(clojure-mode lsp-mode cider lsp-treemacs flycheck company dap-mode))
+
+(when (cl-find-if-not #'package-installed-p package-selected-packages)
+  (package-refresh-contents)
+  (mapc #'package-install package-selected-packages))
+
+(add-hook 'clojure-mode-hook 'lsp)
+(add-hook 'clojurescript-mode-hook 'lsp)
+(add-hook 'clojurec-mode-hook 'lsp)
+
+(setq gc-cons-threshold (* 100 1024 1024)
+      read-process-output-max (* 1024 1024)
+      treemacs-space-between-root-nodes nil
+      company-minimum-prefix-length 1
+      lsp-lens-enable t
+      lsp-signature-auto-activate nil)
+
+(with-eval-after-load 'dap-mode
+  (require 'dap-chrome))
+```
+
+# Project creation
+
+For this recipe, we will use [shadow-cljs](https://github.com/thheller/shadow-cljs).
+
+* Install `npx`
+
+``` bash
+$ npm install -g npx
+```
+
+* Create the sample project
+``` bash
+$ npx create-cljs-project dap-debug-app
+npx: installed 1 in 1.764s
+shadow-cljs - creating project: /home/yyoncho/Sources/dap-debug-app
+Creating: /home/yyoncho/Sources/dap-debug-app/package.json
+Creating: /home/yyoncho/Sources/dap-debug-app/shadow-cljs.edn
+Creating: /home/yyoncho/Sources/dap-debug-app/.gitignore
+Creating: /home/yyoncho/Sources/dap-debug-app/src/main
+Creating: /home/yyoncho/Sources/dap-debug-app/src/test
+----
+Installing shadow-cljs in project.
+
+npm notice created a lockfile as package-lock.json. You should commit this file.
++ shadow-cljs@2.11.23
+added 100 packages from 106 contributors and audited 100 packages in 4.831s
+
+3 packages are looking for funding
+  run `npm fund` for details
+
+found 0 vulnerabilities
+```
+
+# Emacs
+
+Create and open file `src/main/dap/frontend/app.cljs` and make sure that `clojure-lsp` language
```

Comment:
`clojure-lsp` is not required for dap-mode, right?

Reply:
dap-mode needs lsp-mode to be running in the files with breakpoints.

Reply:
Oh, I didn't know about that